Save default view parameters

brew opened the pull request:
This PR allows dataset owners to set up and save a default view for their packages. It uses a new `/package/update_params` os-conductor endpoint to save current view parameters to the package model (https://github.com/openspending/os-conductor/pull/31).

<img width="1274" alt="screen shot 2018-04-12 at 10 54 09" src="https://user-images.githubusercontent.com/3993/38670160-ed1af9be-3e3f-11e8-9cba-d97aa22dd03b.png">

Best to review both this and the [related os-conductor PR](https://github.com/openspending/os-conductor/pull/31) together. These changes are currently live on the staging server: https://staging.openspending.org/.

Closes of openspending/openspending#1187

brew committed:
Fix variable typo

brew committed:
Add isOwner property to package.meta

brew committed:
Add Nodemon and related dev start command script

brew committed:
Add dev-watch build script command

brew committed:
Reset and Clear events for visualisations

brew committed:
Set default parameters is available on package model

brew committed:
Don't use defaultParams if visualizations defined in query string

Only use defaultParams if there are no visualizations defined. This
allows users to setup their own params and share the resultant url.

brew committed:
Downloader service options to bypass local and allow fetch options

brew committed:
Default paramater controls for views

Adds controls for dataset owner to save current view as default. Add
user controls to reset the entire view, or reset the view to the default
parameters.

- Adds a `param-controls` directive to manage the UI and controller code
for the default params.
- View params are saved via the data-package-api to the package model in
the ElasticSearch instance. This is mediated by os-conductor by the new
`/package/update_params` endpoint.
- `confirmation-popover` directive added to UI for the 'Save View As
Default' action, which is only shown to the dataset owner.

vitorbaptista reviewed (approved):
I haven't ran it locally, but tried on staging and it's working great :+1: I would just add a few tests, and change `Save View As Default` to `Save View as Default`. Pretty good when the only suggestion in a review is to lowercase a word, huh? :smile:

brew committed:
Send param as string value.

Because Elasticsearch update merges object values with existing values,
we need a way to replace the params entirely. We don't have access to
Elasticsearch scripting, so the easiest way is to dump the params object
to a string, and send it as a value of a 'params' key. The stringified
json object is parsed back to an object before use.

brew committed:
Lower case 'as'

brew committed:
Fix tests
